### DISCUSSION TITLE
Can Aspire Applications Be Deployed on Self-Hosted or Private Clouds, or Are They Azure-Specific?

### DISCUSSION THREAD

**Discussion by aloksharma1** [microsoft/aspire#6847] [Q&A] [answered]
Hi,

I have been exploring the Aspire framework for .NET applications and was curious about the deployment options available. Specifically:

Is it possible to deploy Aspire-based applications on self-hosted servers or private cloud infrastructures (on-premises solutions)?
Or is Aspire tightly integrated with Azure, making it Azure-specific for deployment?
If self-hosted/private cloud deployments are possible, are there any specific configurations, tools, or guidelines that I should be aware of to ensure compatibility?

Thank you in advance for any insights or resources you can provide!

**Comment by davidfowl** [comment]
Aspire isn’t tightly coupled to azure, we (Microsoft) build tooling for azure. If you use the azure packages then you are coupling yourself to azure and we provide tools to help you dev and deploy to that environment.

There are other tools (like aspirate) https://github.com/prom3theu5/aspirational-manifests that can target some other environments. If we’re successful, other people will build tools for various environments to automate end to end deployment.

With all of that said, you don’t need to wait for automated tools to deploy your application elsewhere. You can deploy manually to any environment before aspire existed by providing the right set of configuration values and environment to various places (that’s what these deployment tools attempt to automate).

**Reply by aloksharma1** [reply]
Hi,

Thanks for your valuable reply, what got me interested in aspire is the abiltiy to turn on/off and trace the web api modules, is there any guide/walkthrough available through which i can deploy or develop with my own build tooling with aspire.

i will look into the link you have shared. Thanks again.

**Reply by davidfowl** [reply]
Are you asking about the dashboard specifically?

**Reply by aloksharma1** [reply]
yes, but for me the important parts right now are resources, structured and traces.

**Comment by davidfowl** [comment, marked as answer]
See the improvements to deployment in the latest https://learn.microsoft.com/en-us/dotnet/aspire/whats-new/dotnet-aspire-9.2#-deployment-improvements
